packages/cysync-core/src/dialogs/HistoryDialog.tsx

```diff
@@ -103,12 +105,17 @@ const selector = createSelector(
   }),
 );
 
-export const HistoryDialog: FC<IHistoryDialogProps> = ({ txn }) => {
+export const HistoryDialog: FC<IHistoryDialogProps> = ({ txn: _txn }) => {
   const { lang, wallets, accounts, priceInfos, isDiscreetMode } =
     useAppSelector(selector);
   const keys = lang.strings.history.dialogBox;
   const dispatch = useAppDispatch();
   const theme = useTheme();
+  const txn = useAppSelector(selectTransactionById(_txn.__id));
+
+  if (txn === undefined) {
+    return <LoaderDialog />;
+  }
 
   const displayTransaction = useMemo(
```

Comment:
Don't call a hook after a conditional return statement. Refer: https://react.dev/reference/rules/rules-of-hooks

Reply:
fixed